.changeset/real-wombats-fly.md

```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': minor
+---
+
+Added ability to change positioning of `Dialog` with `data-iui-placement` in React.
```

Comment:
changesets should mention user-facing apis (`placement` prop) rather than implementation details (`data-iui-placement` attribute)

```suggestion
Added `placement` prop to `Dialog` to allow placing it at one of the corners.
```